snapcraft/_store.py

```diff
@@ -100,6 +101,119 @@ def _requires_login():
         raise
 
 
+@contextmanager
+def _requires_command_success(command, not_found_err=None, failure_err=None):
```

Comment:
This is similar to what Joe did for cloning repos. Search for MissingPackageError.
It's not the same, because this one actually tries the command, but maybe something can be shared?